Branch-SDK/src/io/branch/referral/DeferredAppLinkDataHandler.java

```diff
@@ -0,0 +1,76 @@
+package io.branch.referral;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.text.TextUtils;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+/**
+ * Created by sojanpr on 5/19/16.
+ * Class for handling deferred app links
+ */
+class DeferredAppLinkDataHandler {
+    private static final String NATIVE_URL_KEY = "com.facebook.platform.APPLINK_NATIVE_URL";
+
+    public static Boolean fetchDeferredAppLinkData(Context context, final AppLinkFetchEvents callback) {
+        boolean isRequestSucceeded = true;
+        try {
+            // Init FB SDK
+            Class<?> FacebookSdkClass = Class.forName("com.facebook.FacebookSdk");
+            Method initSdkMethod = FacebookSdkClass.getMethod("sdkInitialize", Context.class);
+            initSdkMethod.invoke(null, context);
```

Comment:
Did you find that this was required? I think I was able to get it to work without calling init on the FB SDK.